core/pkg/eval/ievaluator.go

```diff
@@ -10,14 +12,41 @@ type AnyValue struct {
 	Variant string
 	Reason  string
 	FlagKey string
+	Error   error
 }
 
-func NewAnyValue(value interface{}, variant string, reason string, flagKey string) AnyValue {
+func NewAnyValue(value interface{}, variant string, reason string, flagKey string, err error) AnyValue {
 	return AnyValue{
 		Value:   value,
 		Variant: variant,
 		Reason:  reason,
 		FlagKey: flagKey,
+		Error:   err,
+	}
+}
+
+func (a AnyValue) GetMetricsAttributes() []attribute.KeyValue {
+	if a.Error != nil {
+		return []attribute.KeyValue{
+			semconv.FeatureFlagProviderName("flagd"),
+			{
+				Key:   attribute.Key("reason"),
+				Value: attribute.StringValue(a.Reason),
+			},
+			{
+				Key:   attribute.Key("errType"),
```

Comment:
there is a semconv for errors: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/semantic_conventions/exceptions.md